### PR TITLE
Interactive add creds for lxd

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following adds changes so that you can add certs interactively
for the lxd setup. In addition to add a cloud interactively.

## QA steps

Follow the work flow with the following interactive commands:

```sh
juju add-cloud
```
```sh
juju add-credential lxd
```

## Documentation changes

[demo](https://asciinema.org/a/189669?autoplay=1&speed=3)

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/Epc2WbIN/292-add-verify-interactive-add-cloud-command-for-lxd-cluster
https://trello.com/c/EYCsJfkS/293-add-verify-interactive-add-credential-lxd-cluster

File: provider/lxd/credentials.go [hunk 1]
```diff
@@ -166,27 +171,19 @@ func (p environProviderCredentials) readOrGenerateCert(logf func(string, ...inte
 
 // FinalizeCredential is part of the environs.ProviderCredentials interface.
 func (p environProviderCredentials) FinalizeCredential(ctx environs.FinalizeCredentialContext, args environs.FinalizeCredentialParams) (*cloud.Credential, error) {
-	var interactive bool
+	// TODO (stickupkid): if there are no server certs for the following, use
+	// the lxd API
 	switch authType := args.Credential.AuthType(); authType {
 	case interactiveAuthType:
```
Comment (manadart) [inline provider/lxd/credentials.go]: I don't think this is quite right.

I think the options presented to the user should be:
- certificate
- certificate from files

Then when creating the credential from files, the actual certs are written as a credential of type "certificate".

Comment (SimonRichardson) [reply provider/lxd/credentials.go]: So I've changed this in https://github.com/juju/juju/pull/8879/commits/a14f2ee085e213e9786bdfce2af90deff98e53a6

**Comment by manadart** [comment]
As I mentioned on , we should get that landed, then merge into this one for a look.

I am getting this error when I try to kill a controller I bootstrapped with new credentials:
```
ERROR getting controller environ: getting environ using bootstrap config from client store: tls: failed to find any PEM data in certificate input
15:33:49 DEBUG cmd supercommand.go:459 error stack:
tls: failed to find any PEM data in certificate input
github.com/juju/juju/container/lxd/connection.go:111:
github.com/juju/juju/container/lxd/server.go:77:
github.com/juju/juju/provider/lxd/environ_raw.go:42:
github.com/juju/juju/provider/lxd/environ.go:100:
github.com/juju/juju/provider/lxd/provider.go:118:
github.com/juju/juju/cmd/juju/controller/destroy.go:537: getting environ using bootstrap config from client store
github.com/juju/juju/cmd/juju/controller/kill.go:121: getting controller environ
```

But this stack trace will be all different once we're using that code.

**Comment by SimonRichardson** [comment]
@manadart Awesome, agreed!

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**